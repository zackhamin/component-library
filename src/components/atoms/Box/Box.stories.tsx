import { colors } from 'Styles';
import { Box, BoxProps } from './Box';

export default {
	title: 'atoms/Box',
	component: Box,
};

export const BoxStory = (args: BoxProps) => <Box {...args} />;

BoxStory.storyName = 'Box';
BoxStory.args = {
	backgroundColor: colors.darkGrey,
	padding: '2.4rem',
	children: 'I am a box',
};
