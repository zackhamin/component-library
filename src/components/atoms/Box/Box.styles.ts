import styled from 'styled-components';

const StyledBox = styled.div<{
	backgroundColor: string;
	margin: string;
	padding: string;
}>`
	background-color: ${({ backgroundColor }) => backgroundColor};
	border-radius: 0.8rem;
	margin: ${({ margin }) => margin};
	padding: ${({ padding }) => padding};
`;

export { StyledBox };
