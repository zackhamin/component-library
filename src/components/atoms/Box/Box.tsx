import { colors } from 'Styles';
import { ReactNode } from 'react';
import { StyledBox } from './Box.styles';

export interface BoxProps {
	backgroundColor?: string;
	// not typing these as `Size` to allow for multiple values, e.g. `0 1.6rem 2.4rem 0
	margin?: string;
	padding?: string;
	children: ReactNode;
}

const Box = ({
	backgroundColor = colors.white,
	margin = '0',
	padding = '1.6rem',
	children,
}: BoxProps) => {
	return (
		<StyledBox
			backgroundColor={backgroundColor}
			padding={padding}
			margin={margin}
		>
			{children}
		</StyledBox>
	);
};

export { Box };
