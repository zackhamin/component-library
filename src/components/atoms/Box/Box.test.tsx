import { Box } from '../Box';
import { render } from '@testing-library/react';

describe('Box', () => {
	it('renders correctly', () => {
		const { container } = render(<Box>I am a box</Box>);
		expect(container.firstChild).toMatchSnapshot();
	});
});
