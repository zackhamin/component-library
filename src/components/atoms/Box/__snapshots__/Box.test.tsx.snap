// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders correctly 1`] = `
.c0 {
  background-color: #ffffff;
  border-radius: 0.8rem;
  margin: 0;
  padding: 1.6rem;
}

<div
  class="c0"
>
  I am a box
</div>
`;
